File: app/src/main/java/com/anadi/weatherinfo/addlocation/AddLocationContract.java
package com.anadi.weatherinfo.addlocation;

import androidx.annotation.StringRes;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Observer;

public interface AddLocationContract {

    interface View {

        void loading();

        void onError(@StringRes int resId);

        void onAddedSuccess();

        void updateCityList(@NotNull ArrayList<String> cities);
    }

    interface Presenter {
        void addLocation(final String cityName, final String countryName);

        void onCountrySelected(final String countryName);

        ArrayList<String> getCountryNames();
    }

    interface Model {
        boolean add(String cityName, String countryName);
    }
}

File: app/src/main/java/com/anadi/weatherinfo/addlocation/AddLocationContract.kt
package com.anadi.weatherinfo.addlocation

import androidx.annotation.StringRes
import java.util.*

interface AddLocationContract {
    interface View {
        fun loading()
        fun onError(@StringRes resId: Int)
        fun onAddedSuccess()
        fun updateCityList(cities: ArrayList<String?>)
    }

    interface Presenter {
        fun addLocation(cityName: String?, countryName: String?)
        fun onCountrySelected(countryName: String?)
        val countryNames: ArrayList<String?>?
    }

    interface Model {
        fun add(cityName: String?, countryName: String?): Boolean
    }
}